controllers/profiles/common/ensurer.go

```diff
@@ -36,6 +36,9 @@ var _ ObjectEnsurer = &noopObjectEnsurer{}
 type ObjectEnsurer interface {
 	Ensure(ctx context.Context, workflow *operatorapi.SonataFlow, visitors ...MutateVisitor) (client.Object, controllerutil.OperationResult, error)
 }
+type ObjectEnsurerWithPlatform interface {
```

Comment:
@wmedvede not all ensurers request platform. It's bad adding this unused parameter in some cases just because the API requires it. That's why we opted to add a new interface.